### PR TITLE
fix(ui): fix checkbox toggle and consolidate implementations

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary

Checkboxes weren't properly toggling due to a broken indeterminate state mapping. This PR fixes the root cause and consolidates all checkbox implementations to use the central design system component.

**Root cause:** `TanStackHeaderCheckbox` passed `checked={false}` when state was indeterminate. The separate `indeterminate` prop tried to set a DOM property on a `<button>` element (no-op, only works on `<input type="checkbox">`). Radix only reads its React `checked` prop, so the checkbox rendered as unchecked.

**Fixes:**
- Base Checkbox now renders Minus icon for indeterminate state, Check icon for checked
- `TanStackHeaderCheckbox` passes `checked="indeterminate"` directly to Radix
- `CreatorProfileTableRow` checkbox container now properly centered (was top-left)
- Removed `aria-hidden="true"` from interactive checkbox wrappers (screen reader fix)
- `AudienceRowSelectionCell` aligned to same wrapper pattern as `TanStackRowCheckbox`
- Removed ad-hoc className overrides from admin checkboxes (consistent design system usage)

## Test Coverage

Tests: 28 → 29 (+1 new)
- 7 indeterminate state tests (up from 2), including click interaction test
- All 29 checkbox unit tests pass

## Pre-Landing Review

No issues found. Pure UI component changes, no SQL/auth/LLM/shell surface.

## Adversarial Review

Claude adversarial found 7 items, all low severity or pre-existing. One valid edge case (deprecated `indeterminate` prop without `checked` creates controlled mode) mitigated by deprecation + no current consumers using that pattern.

## Test plan
- [x] All checkbox unit tests pass (29/29)
- [x] Pre-commit hooks pass (biome, eslint, a11y, typecheck)
- [x] Pre-push hooks pass (biome lint, tailwind, typecheck)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed checkbox indeterminate state behavior for proper "select all" functionality
  * Corrected checkbox alignment in admin profile table rows
  * Improved checkbox accessibility for screen readers

* **Changed**
  * Updated all table checkboxes to use a standardized design-system component across admin, audience, and dashboard tables
  * Indeterminate checkboxes now display with a visual indicator (minus icon)

* **Version**
  * Released version 26.4.143.1

### CODE REVIEW

File: packages/ui/atoms/checkbox.tsx [hunk 1]
```diff
@@ -68,7 +51,11 @@ const Checkbox = React.forwardRef<
       <CheckboxPrimitive.Indicator
         className={cn('flex items-center justify-center text-current')}
       >
-        <Check className='h-3 w-3 [stroke-width:2.5]' />
+        {effectiveChecked === 'indeterminate' ? (
+          <Minus className='h-3 w-3 [stroke-width:2.5]' />
+        ) : (
+          <Check className='h-3 w-3 [stroke-width:2.5]' />
+        )}
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply packages/ui/atoms/checkbox.tsx:59]: Fixed in 075f6d450. Added clarifying comment documenting that icon selection uses prop-based `effectiveChecked` rather than internal Radix state.
